homeassistant/components/asuswrt.py

```diff
@@ -0,0 +1,67 @@
+"""
+Support for ASUSWRT devices.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/asuswrt/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.const import (
+    CONF_HOST, CONF_PASSWORD, CONF_USERNAME, CONF_PORT, CONF_MODE,
+    CONF_PROTOCOL)
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.discovery import async_load_platform
+
+REQUIREMENTS = ['aioasuswrt==1.1.3']
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = "asuswrt"
+DATA_ASUSWRT = DOMAIN
+
+CONF_PUB_KEY = 'pub_key'
+CONF_SSH_KEY = 'ssh_key'
+CONF_REQUIRE_IP = 'require_ip'
+DEFAULT_SSH_PORT = 22
+SECRET_GROUP = 'Password or SSH Key'
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_HOST): cv.string,
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Optional(CONF_PROTOCOL, default='ssh'): vol.In(['ssh', 'telnet']),
+        vol.Optional(CONF_MODE, default='router'): vol.In(['router', 'ap']),
+        vol.Optional(CONF_PORT, default=DEFAULT_SSH_PORT): cv.port,
+        vol.Optional(CONF_REQUIRE_IP, default=True): cv.boolean,
+        vol.Exclusive(CONF_PASSWORD, SECRET_GROUP): cv.string,
+        vol.Exclusive(CONF_SSH_KEY, SECRET_GROUP): cv.isfile,
+        vol.Exclusive(CONF_PUB_KEY, SECRET_GROUP): cv.isfile
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+
+async def async_setup(hass, config):
+    """Set up the asuswrt component."""
+    from aioasuswrt.asuswrt import AsusWrt
+
+    conf = config[DOMAIN]
+    if conf[CONF_HOST] is None or conf[CONF_USERNAME] is None:
```

Comment:
I was thinking we should call the api with the credentials and see if we can login. If that fails we should log error return False.

Config has already been validated, so we don't need to do that here.

Reply:
This is a bit of a hassle, it makes no calls in init at all atm. Ofc we could do a login, but it will probably not raise an exception (if the network goes down it would flood the logs in hass)... Ill think about how to implement this.